apm-sniffer/apm-sdk-plugin/underdow-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/undertow1x/UndertowInvokeInterceptor.java

```diff
@@ -0,0 +1,112 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.apache.skywalking.apm.plugin.undertow1x;
+
+import io.undertow.server.HttpServerExchange;
+import io.undertow.util.HeaderMap;
+import org.apache.skywalking.apm.agent.core.context.CarrierItem;
+import org.apache.skywalking.apm.agent.core.context.ContextCarrier;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceConstructorInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+
+import javax.servlet.DispatcherType;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Copyright @ 2018/7/20
+ *
+ * @author cloudgc
+ */
+public class UndertowInvokeInterceptor implements InstanceMethodsAroundInterceptor, InstanceConstructorInterceptor {
+
+    private static final String FORWARD_REQUEST_FLAG = "SW_FORWARD_REQUEST_FLAG";
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
+
+        HttpServerExchange exchange = (HttpServerExchange) allArguments[0];
+
+        DispatcherType dispatcherType = (DispatcherType) allArguments[3];
+
+        if (dispatcherType == DispatcherType.FORWARD) {
+            if (ContextManager.isActive()) {
+                AbstractSpan abstractTracingSpan = ContextManager.activeSpan();
+                Map<String, String> eventMap = new HashMap<String, String>();
+                eventMap.put("forward-url", exchange.getRequestURI());
+                abstractTracingSpan.log(System.currentTimeMillis(), eventMap);
+                ContextManager.getRuntimeContext().put(FORWARD_REQUEST_FLAG, true);
```

Comment:
You must clear the stuff that you put in RuntimeContext. or it will casue the memory leak.